httpbp/client_middlewares.go

```diff
@@ -0,0 +1,225 @@
+package httpbp
+
+import (
+	"errors"
+	"io"
+	"net/http"
+	"sync"
+	"sync/atomic"
+
+	retry "github.com/avast/retry-go"
+	"github.com/opentracing/opentracing-go"
+
+	"github.com/reddit/baseplate.go/breakerbp"
+	"github.com/reddit/baseplate.go/retrybp"
+	"github.com/reddit/baseplate.go/tracing"
+)
+
+const (
+	defaultMaxErrorReadAhead = 1024
```

Comment:
We usually export our default constants, this way in the config struct we can point the users to the default value which would appear on pkg.go.dev.